src/main/java/com/moneylendingapp/security/jwt/JwtRequestFilter.java

```diff
@@ -0,0 +1,61 @@
+package com.moneylendingapp.security.jwt;
+
+import com.moneylendingapp.exceptions.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+    private final UserDetailsService userDetailsService;
+    private final JwtUtil jwtUtil;
+    private final JwtConfig jwtConfig;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) {
+        try {
+
+            final String authorizationHeader = request.getHeader(jwtConfig.getAuthorizationHeader());
+
+            String username = null;
```

Comment:
You're making it possible to have nullpointer. why not use Optional.empty() here first and then fill it up when you find the value